contributing.md

```diff
@@ -162,10 +162,21 @@ get acquainted with this development process.
 
 1. **Choose a base branch.**
    - If your changes will break API or ABI, then base your new branch off of `main`.
+     If this package's `main` branch is not part of the upcoming Gazebo collection,
```

Comment:
Might not be easy but I think that we should drive to simplify rules so people that read "Choose a base branch" with the intention of coding as soon as possible don't fall into too much effort to start that. This `main` branch procedure that is well described seems something more designed for the core team than for the newcomers. 

For newcomers, different difficulties are here: what is a collection? Which is the upcoming collection nowadays? How can I check if main branch is part of any collection? Why I need to create an issue if all I what to do is to send a PR?.

Reply:
I created a proposal here about changing our branch naming scheme in the hopes of making it simpler for users: https://github.com/gazebo-tooling/release-tools/issues/1273